src/Microsoft.DotNet.Interactive.HttpRequest/HttpRequestKernel.cs

```diff
@@ -106,25 +107,23 @@ async Task IKernelCommandHandler<SubmitCode>.HandleAsync(SubmitCode command, Ker
 
         try
         {
-            foreach (var parsedRequest in parsedRequests)
+            foreach (var requestMessage in requestMessages)
             {
-                await HandleRequestAsync(parsedRequest, command, context);
+                if (requestMessage is not null)

```

Comment:
consider: Move this into the `Select` lambda on line 97 (and only select results that have `BindingResult.IsSuccessful` set to `true`)

Reply:
The main reason for the `if not null` check was to avoid the nullability warning from the compiler. I agree that moving the check into LINQ code seems cleaner. Doing that in a way that avoids the warning seems a bit uglier. All highly subjective and I don't have a strong preference beyond correctness here.

Here's what I've got. (Not the `r.Value!` is needed).

```csharp

        var requestMessages = httpRequestResults
                              .Where(r => r is { IsSuccessful: true, Value: not null })
                              .Select(r => r.Value!).ToArray();

        try
        {
            foreach (var requestMessage in requestMessages)
            {
                await SendRequestAsync(requestMessage, command, context);
            }
        }

```

Reply:
We could use the nullable annotation on `.Value` to indicate that it can never be `null` when `IsSuccessful` is `true`. But I think the annotations are not universally available so we would need to add a source file that declares those attributes in the netstandard2.0 case (and make sure that similar code is also available in the other repos where the source package is consumed).

Reply:
> We could use the nullable annotation on .Value to indicate that it can never be null when IsSuccessful is true.

`null` is a valid value.